README.md

```diff
@@ -32,6 +32,89 @@ flumedb.append({foo: 'bar'}, function (err) {
 })
 ```
 
+## API
+
+### `FlumeviewLevel(version, map) => function`
+
+#### `version`
+The version of the view. Incrementing this number will cause the view to be re-built
+
+#### `map`
+A function with signature `(value, seq)`, where `value` is the item from the log coming past, and `seq` is the location of that value in the flume log.
+
+This function **must return an Array** that's either empty or contains unique index key(s).
+These index keys can then be queired to retrieve the stored value (see `get` and `read` below).
+
+Examples of index key(s) you might return:
+- `[]` - i.e. don't add any indexes for this `value`
+- `['@mix']` - make an index entry for this value under string `@mix`
+- `['@mix', '@mixmix']` - make an index entries for this value under both `@mix` AND `@mixmix`
+- `[['@mix', 1524805117433]]` - make an index entry for this value under the key `['@mix', 1524805117433]` (anything can be a key in leveldb)
+
+This last case is useful when you might want multiple entries under a particular key like `@mix` - if just use `@mix` then the index will get overwritten by future values coming in with the same key.
+Extending the key to include some unique aspect (like a timestamp or the `seq` of the value) means you can have multiple indexes in your view which have a _similar_ key.
+
+e.g. [flumeview-search](https://github.com/flumedb/flumeview-search) is a flumeview which takes the text from incoming values and builds an index which can be searched.
+It takes a sentence like "Learn about leveldb" and maps that into index keys like `[['learn', -145], ['about', -145], ['leveldb', -145]]`, meaning we can later search "leveldb" and find this sentence, as well as all other sentences also containing "leveldb".
+
+
+#### `function`
+flumeview-level returns a function which follows the flumeview pattern, enabling it to be installed into a flumedb.
+
+
+### `get(key, cb)`
+
+This is a method that gets attached to the flumedb after you install your flumeview (see example above).
+
+The keys for the values in `map` above would be `'@mix'`, `'@mixmix'`, or `['@mix', 1524805117433]`
+
+
+### `read(opts) => pull-stream`
+
+`opts` is similar to a level db query ([see level docs](https://github.com/Level/levelup#dbcreatereadstreamoptions)).
+
+e.g.
+
+```js
+{
+  live: true,     // this is an addition to the classic query options of level
+  gte: '@mi',     // gte = greater than or equal to
+  lt: null,       // lt = less than
+  reverse: true,
+  keys: true,
+  values: true,
+  seqs: false,
+}
+```
+
+If you've created indexes that are Arrays, have a read of how Arrays and other value are ordered [here](https://github.com/deanlandolt/bytewise#order-of-supported-structures) (flumeview-level uses [charwise](https://github.com/dominictarr/charwise) for ordering, but this is based on [bytewise](https://github.com/deanlandolt/bytewise))
```

Comment:
Why do we need to care about the ordering of indexes in arrays?

Reply:
the array IS the index (key)... and ordering matters because level lets you do queries using bounds ( >, >=, <=, < ), so you have to understand how the arrays are going to be ordered in order to do queries ...

Reply:
@pietgeursen using leveldb is all about ordering keys so that you can do queries efficiently. because of the way a log-structured-merge-tree works (what level is) it can read adjacent records quickly (with a single seek) but jumping around is not as fast.

Reading M records from a database containing N items is probably something like be log(N)+M to stream, but would be log(N)*M to jump around. (log(N) is the lookup cost, a stream needs just one lookup...). Thats rough, back of the napkin of course.